src/sentry/api/endpoints/organization_member/__init__.py

```diff
@@ -1,28 +1,86 @@
 from __future__ import annotations
 
-from typing import Collection
+from typing import Collection, TypedDict
 
 from django.db import transaction
 from rest_framework.request import Request
 
 from sentry import roles
+from sentry.api.exceptions import SentryAPIException, status
 from sentry.auth.superuser import is_active_superuser
-from sentry.models import Organization, OrganizationMember, OrganizationMemberTeam
-from sentry.roles.manager import Role
+from sentry.models import Organization, OrganizationMember, OrganizationMemberTeam, Team, TeamStatus
+from sentry.roles.manager import Role, TeamRole
+
+
+class InvalidTeam(SentryAPIException):
+    status_code = status.HTTP_400_BAD_REQUEST
+    code = "invalid_team"
+    message = "The team slug does not match a team in the organization"
+
+
+class TeamRoleDict(TypedDict):
+    teamSlug: str
+    role: str
 
 
 @transaction.atomic
-def save_team_assignments(organization_member, teams):
-    # teams may be empty
+def save_team_assignments(
+    organization_member: OrganizationMember,
+    teams: list[str],
+    teams_with_roles: list[TeamRoleDict] = None,
+):
+    team_slugs = [item["teamSlug"] for item in teams_with_roles] if teams_with_roles else teams
```

Comment:
`if teams_with_roles` treats `[]` the same as `None`. I'd change it to `if teams_with_roles is not None` (and maybe invert the if-else order).